File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> h1=new HashSet<>();
        int left=0,right=0,max=0;
        while(right<s.length())
        {
            if(!h1.contains(s.charAt(right)))
            {
                h1.add(s.charAt(right));
                max=Math.max(max,right-left+1);
                right++;
            }
            else
            {
                h1.remove(s.charAt(left));
                left++;
            }
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let left=0,right=0;
    const h1=new Set();
    let Max=0;
    while(right<s.length)
    {
        if(!h1.has(s[right]))
        {
            h1.add(s[right]);
            Max=Math.max(Max,right-left+1);
            right++;
        }
        else
        {
            h1.delete(s[left]);
            left++;
        }
    }
    return Max;
};